File: Array/Count of divisors.go
package main

import (
	"fmt"
)

func main() {

	arr := []int{8, 9, 10}
	result := countDivisors(arr)
	fmt.Println(result)
}

func countDivisors(A []int) []int {

	var divisors [1000005]int
	var ansArr []int

	for i := 1; i <= 1000000; i++ {
		divisors[i] = i
	}

	for i := 2; i <= 1000000; i++ {
		if divisors[i] == i {
			for j := i * i; j <= 1000000; j += i {
				if divisors[j] == j {
					divisors[j] = i
				}
			}
		}
	}

	for i := 0; i < len(A); i++ {
		temp := A[i]
		ans := 1

		for temp != 1 {
			count := 1
			div := divisors[temp]
			for temp != 1 && temp%div == 0 {
				count++
				temp /= div
			}
			ans *= count
		}

		ansArr = append(ansArr, ans)
	}

	return ansArr
}

File: Pythons Solutions/Count of divisors.py
class Solution:
    def countDivisors(self, arr):
        n = 1000005
        divisor = [0]*n

        for i in range(n):
           divisor[i] = i
        
        for i in range(2, n):
            if divisor[i] == i:
                j = i*i
                while j <n:
                    if divisor[j] == j:
                        divisor[j] = i
                    j += i
        
        solution = []

        for i in range(len(arr)):
            temp = arr[i]
            ans = 1
            while temp != 1:
                count = 1
                div = divisor[temp]
                while temp != 1 and temp%div == 0:
                    count += 1
                    temp = temp // div
                
                ans *= count
            solution.append(ans)
        
        return solution

       

obj = Solution()
arr = [2, 2, 3, 2]
ans = obj.countDivisors(arr)
print(ans)